src/test/java/uk/gov/companieshouse/service/AppealServiceTest.java

```diff
@@ -0,0 +1,132 @@
+package uk.gov.companieshouse.service;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.companieshouse.exception.AppealNotFoundException;
+import uk.gov.companieshouse.model.Appeal;
+import uk.gov.companieshouse.model.CreatedBy;
+import uk.gov.companieshouse.model.OtherReason;
+import uk.gov.companieshouse.model.PenaltyIdentifier;
+import uk.gov.companieshouse.model.Reason;
+import uk.gov.companieshouse.repository.AppealRepository;
+
+import java.util.Optional;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AppealServiceTest {
+
+    private static final String TEST_COMPANY_ID = "12345678";
+    private static final String TEST_RESOURCE_ID = "1";
+    private static final String TEST_PENALTY_REFERENCE = "A12345678";
+    private static final String TEST_REASON_TITLE = "This is a title";
+    private static final String TEST_REASON_DESCRIPTION = "This is a description";
+    private static final String TEST_ERIC_ID = "1";
+
+    @InjectMocks
+    private AppealService appealService;
+
+    @Mock
+    private AppealRepository appealRepository;
+
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void testCreateAppeal_returnsResourceId() throws Exception {
+
+        Appeal appeal = createAppeal();
+
+        Appeal persistedAppeal = createAppeal();
+
+        persistedAppeal.setCreatedBy(new CreatedBy(TEST_ERIC_ID));
+        persistedAppeal.setId(TEST_RESOURCE_ID);
+        when(appealRepository.insert(any(Appeal.class))).thenReturn(persistedAppeal);
+
+        String resourceId = appealService.saveAppeal(TEST_COMPANY_ID, appeal, TEST_ERIC_ID);
+
+        assertThat(resourceId, is(notNullValue()));
+        assertThat(resourceId, is(TEST_RESOURCE_ID));
+
+    }
+
+    @Test
+    public void testCreateAppeal_throwsExceptionIfNoResourceIdReturned() throws Exception {
```

Comment:
Is that scenario even real? It simulates that repository returned appeal without ID when persistence failed? Is that how repository really behaves? As a developer I'd rather expect repository either throw an exception or return null in worst case scenario. Returning input on failure seems unreal to me?

Have you checked how repository really behaves?

Reply:
the crud repository returns an Optional<T> for findById() - i have therefore changed how our service handles this